lib/dry/transaction/dsl.rb

```diff
@@ -26,7 +26,7 @@ def define_dsl
         module_exec(@step_adapters) do |step_adapters|
           step_adapters.each do |adapter_name, adapter|
             define_method(adapter_name) do |step_name, with: nil, **options|
-              operation_name = with || step_name
+              operation_name = with
```

Comment:
@timriley I think it would make more readable the code if we actually use something like `source` for the `Step`

```ruby
source = with ? :external : :internal
operation_name = with || step_name
```

That way the method `internal?` and `external?` are much better to reason about. 

https://github.com/dry-rb/dry-transaction/pull/109/files#diff-503c8ab3d6cf49b4e18fe227885fca0fR78

Reply:
@GustavoCaso Thanks for the suggestion, but I'm not sure introducing a new magical symbol really helps with improving the understandability of those methods. If anything, it forces you to go up a layer and work out _why_ that symbol was set.

Either way, I think this will be something we can fully clear with some documentation (both API-level and user-facing docs). I intend to do some work on the user-facing docs to go along with this change being released, too.

Reply:
@timriley Thanks for the comment. 
Totally agree that is something we can work and clear with some documentation, let me know if I can help you with the documentation.